### ISSUE TITLE
it is  a bug?

### ISSUE THREAD

**Issue by toxotguo** [ethereum/cpp-ethereum#3623]
in file libetherenum/BlockChainSync.cpp  function BlockChainSync::requestBlocks 
line 319 

if (start <= 1)
	m_haveCommonHeader = true; //reached genesis

should be if( start <=0 ) ?
otherwise Sync Fail

**Comment by toxotguo** [comment]
@chfast

**Comment by chfast** [comment]
Thanks, @toxotguo,
What do you think @gumb0?

**Comment by gumb0** [comment]
Yeah, looks like `if (start == 0)` would be more correct, but we need to find a way to test it before fixing (create chain reorg soon after sync start probably?)

**Comment by chfast** [comment]
It looks we don't have any tests for BlockChainSync: https://codecov.io/gh/ethereum/cpp-ethereum/src/develop/libethereum/BlockChainSync.cpp

Can we add some unit tests?

@winsvega is it possible to change that?

**Comment by winsvega** [comment]
well we could add some unit tests for that. 
but I think that upcoming eth RPC tests would be able to cover this class while running all tests on Azure. 

https://github.com/ethereum/tests/tree/RPCTests/RPCTests
